test/Microsoft.ML.Tests/Scenarios/Api/TestApi.cs

```diff
@@ -183,6 +183,93 @@ public void TrainAveragedPerceptronWithCache()
             Assert.Equal(1, globalCounter);
         }
 
+        [Fact]
+        public void MetadataSupportInDataViewConstruction()
+        {
+            var data = ReadBreastCancerExamples();
+            using (var env = new ConsoleEnvironment(0))
+            {
+                var autoSchema = SchemaDefinition.Create(typeof(BreastCancerExample));
+
+                // Create Metadata.
+                var kindFloat = "Testing float as metadata.";
+                var valueFloat = 10;
+                var coltypeFloat = NumberType.Float;
+                var kindString = "Testing string as metadata.";
+                var valueString = "Strings have value.";
+                var kindStringArray = "Testing string array as metadata.";
+                var valueStringArray = "I really have no idea what these features entail.".Split(' ');
+                var kindFloatArray = "Testing float array as metadata.";
+                var valueFloatArray = new float[] { 1, 17, 7, 19, 25, 0 };
+                var kindVBuffer = "Testing VBuffer as metadata.";
+                var valueVBuffer = new VBuffer<float>(4, new float[] { 4, 6, 89, 5 });
+
+                var metaFloat = new Microsoft.ML.Runtime.Api.MetadataInfo<float>(kindFloat, valueFloat, coltypeFloat);
+                var metaString = new Microsoft.ML.Runtime.Api.MetadataInfo<string>(kindString, valueString);
+
+                // Add Metadata.
+                var labelColumn = autoSchema[0];
+                var labelColumnWithMetadata = new SchemaDefinition.Column(env, labelColumn.MemberName, labelColumn.ColumnType,
+                    metadataInfos: new Microsoft.ML.Runtime.Api.MetadataInfo[] { metaFloat, metaString });
+
+                var featureColumnWithMetadata = autoSchema[1];
+                featureColumnWithMetadata.AddMetadata(kindStringArray, valueStringArray);
+                featureColumnWithMetadata.AddMetadata(kindFloatArray, valueFloatArray);
+                featureColumnWithMetadata.AddMetadata(kindVBuffer, valueVBuffer);
+
+                var mySchema = new SchemaDefinition { labelColumnWithMetadata, featureColumnWithMetadata };
+                var idv = env.CreateDataView(data, mySchema);
+
+                // Test GetMetadataTypes.
+                var internalSchemaLabelMetadataTypes = idv.Schema.GetMetadataTypes(0).ToArray();
```

Comment:
>GetMetadataTypes [](start = 66, length = 16)

This is new code. Please use the new `Schema` idioms for accessing metadata, not the old `ISchema` idioms. To keep using `ISchema` just adds *more* technical debt we'll have to clean up.

HEre and everywhere.